modules/nf-core/arcashla/extract/main.nf

```diff
@@ -0,0 +1,42 @@
+process ARCASHLA_EXTRACT {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::arcas-hla=0.5.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/arcas-hla:0.5.0--hdfd78af_0':
+        'quay.io/biocontainers/arcas-hla:0.5.0--hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(bam)
+
+    output:
+    tuple val(meta), path("*.fq.gz"), emit: extracted_reads_fastq
```

Comment:
You need to add support for optional output files (`--keep_files`)

Reply:
I guess these intermediate files should also be included in the `test.yml` right? I didn't find a good solution for now since the tool creates an additional folder for the intermediate files that contains an UUID. Do you have a suggestion?

Reply:
Hmm that is challenging. The only thing I can think of is seeing if the pytest `path` allows wildcards and then you can check for presence of a particular file in `--keep-file`? If it doesn't support it I think we can let it slide. That said it looks like from your `input:` the intermediate files are stored in a directory called `temp_files`? chceking just for that directory might be sufficient.